torch/nn/modules/transformer.py

```diff
@@ -80,8 +80,16 @@ def __init__(self, d_model: int = 512, nhead: int = 8, num_encoder_layers: int =
 
         self.d_model = d_model
         self.nhead = nhead
-
+        self.num_encoder_layers = num_encoder_layers
```

Comment:
We often see models changed after the fact.  That raises the question whether we can trust these values after the fact.  Most values can be obtained by introspecting the model.  What would it take to provide methods to do that.?

Reply:
Maybe we should add a property to get those modules